service/history/historyEngine.go

```diff
@@ -2844,8 +2844,10 @@ func (e *historyEngineImpl) ReapplyEvents(
 				Noop:               false,
 				CreateWorkflowTask: true,
 			}
-			// Do not create workflow task when the workflow is cron and the cron has not been started yet
-			if mutableState.GetExecutionInfo().CronSchedule != "" && !mutableState.HasProcessedOrPendingWorkflowTask() {
+			executionInfo := mutableState.GetExecutionInfo()
+			// Do not create workflow task when the workflow has first workflow task backoff and execution is not started yet
+			workflowTaskBackoff := timestamp.TimeValue(executionInfo.GetExecutionTime()).After(timestamp.TimeValue(executionInfo.GetStartTime()))
+			if workflowTaskBackoff && !mutableState.HasProcessedOrPendingWorkflowTask() {
```

Comment:
Shall we create a util function for this check?

Reply:
👍 this is also needed here https://github.com/temporalio/temporal/blob/87c3bebdb4e2a61e6763c62194b33e3967f16dee/service/history/historyEngine.go#L1804

Reply:
will do